go/vt/vtorc/config/config.go

```diff
@@ -42,200 +42,301 @@ const (
 )
 
 var (
-	sqliteDataFile                 = "file::memory:?mode=memory&cache=shared"
-	instancePollTime               = 5 * time.Second
-	snapshotTopologyInterval       = 0 * time.Hour
-	reasonableReplicationLag       = 10 * time.Second
-	auditFileLocation              = ""
-	auditToBackend                 = false
-	auditToSyslog                  = false
-	auditPurgeDuration             = 7 * 24 * time.Hour // Equivalent of 7 days
-	recoveryPeriodBlockDuration    = 30 * time.Second
-	preventCrossCellFailover       = false
-	waitReplicasTimeout            = 30 * time.Second
-	tolerableReplicationLag        = 0 * time.Second
-	topoInformationRefreshDuration = 15 * time.Second
-	recoveryPollDuration           = 1 * time.Second
-	ersEnabled                     = true
-	convertTabletsWithErrantGTIDs  = false
+	instancePollTime = viperutil.Configure(
+		"InstancePollTime",
+		viperutil.Options[time.Duration]{
+			FlagName: "instance-poll-time",
+			Default:  5 * time.Second,
+			Dynamic:  true,
+		},
+	)
+
+	preventCrossCellFailover = viperutil.Configure(
+		"PreventCrossCellFailover",
+		viperutil.Options[bool]{
+			FlagName: "prevent-cross-cell-failover",
+			Default:  false,
+			Dynamic:  true,
+		},
+	)
+
+	sqliteDataFile = viperutil.Configure(
+		"SQLiteDataFile",
+		viperutil.Options[string]{
+			FlagName: "sqlite-data-file",
+			Default:  "file::memory:?mode=memory&cache=shared",
+			Dynamic:  false,
+		},
+	)
+
+	snapshotTopologyInterval = viperutil.Configure(
+		"SnapshotTopologyInterval",
+		viperutil.Options[time.Duration]{
+			FlagName: "snapshot-topology-interval",
+			Default:  0 * time.Hour,
+			Dynamic:  true,
+		},
+	)
+
+	reasonableReplicationLag = viperutil.Configure(
+		"ReasonableReplicationLag",
+		viperutil.Options[time.Duration]{
+			FlagName: "reasonable-replication-lag",
+			Default:  10 * time.Second,
+			Dynamic:  true,
+		},
+	)
+
+	auditFileLocation = viperutil.Configure(
+		"AuditFileLocation",
+		viperutil.Options[string]{
+			FlagName: "audit-file-location",
+			Default:  "",
+			Dynamic:  false,
+		},
+	)
+
+	auditToBackend = viperutil.Configure(
+		"AuditToBackend",
+		viperutil.Options[bool]{
+			FlagName: "audit-to-backend",
+			Default:  false,
+			Dynamic:  true,
+		},
+	)
+
+	auditToSyslog = viperutil.Configure(
+		"AuditToSyslog",
+		viperutil.Options[bool]{
+			FlagName: "audit-to-syslog",
+			Default:  false,
+			Dynamic:  true,
+		},
+	)
+
+	auditPurgeDuration = viperutil.Configure(
+		"AuditPurgeDuration",
+		viperutil.Options[time.Duration]{
+			FlagName: "reasonable-replication-lag",
```

Comment:
The flag name doesn't match the name of the config (discovered it while going through all dynamic vars)

Reply:
Right here is the reason to include every dynamic config in the e2e test. in the viperutil package we can only test that if a config is specified correctly, it is in fact dynamic. Each binary will need tests that make sure that dynamic config has been coded properly.